File: template_frontend/template/src/models/dto/UserDTO.ts
export class UserDTO {
    id: number;
    email: string;
    name: string;
    profilePicture: string|null;
    language: string|null;
    oldPassword: string|null;
    newPassword: string|null;
    token: string|null;
    isLocalPasswordSet: boolean;

    constructor(email: string, name: string, profilePicture: string|null, language: string|null, oldPassword: string|null, newPassword: string|null, token: string|null, id: number, isLocalPasswordSet: boolean) {
        this.email = email;
        this.name = name;
        this.profilePicture = profilePicture;
        this.language = language;
        this.oldPassword = oldPassword;
        this.newPassword = newPassword;
        this.token = token;
        this.id = id;
        this.isLocalPasswordSet = isLocalPasswordSet;
    }
}

export default UserDTO;

File: template_backend/src/main/java/com/benny1611/template/dto/UserDTO.java
package com.benny1611.template.dto;

import com.benny1611.template.util.ValidUserDTOPasswordChange;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.util.Objects;

@Data
@ValidUserDTOPasswordChange
public class UserDTO {
    @NotNull
    private Long id;
    private String profilePicture;
    private String name;
    @Email
    private String email;
    private String language;
    private String oldPassword;
    private String newPassword;
    private String token;
    private boolean isLocalPasswordSet;

    @Override
    public boolean equals(Object o) {
        if (o == null || getClass() != o.getClass()) return false;
        UserDTO userDTO = (UserDTO) o;
        return Objects.equals(id, userDTO.id);
    }
}
